Troubleshooting.md

```diff
@@ -0,0 +1,84 @@
+# Troubleshooting
+
+This document will help troubleshoot the common debugging errors. If you still cannot solve your problems, please [submit an issue](https://github.com/Microsoft/vscode-java-debug/issues).
+
+## Java Language Support extension fails to start.
+Java Debugger is leveraging `"Language Support for Java(TM) by Red Hat"` extension for source mapping and project support. If Java Language Support extension fails to start, Java Debugger may not work as expected. Here is a simple way to check the Language Server is started. Open a Java or pom.xml file, and then check the status icon at the right corner of VS Code status bar is thumbs up or not.
+
+ ![ status indicator ](https://raw.githubusercontent.com/redhat-developer/vscode-java/master/images/statusMarker.png).
+
+### Try:
+1. If you get the error `"The JAVA_HOME environment variable points to a missing folder"` or `"Java runtime could not be located"`, please make sure the JAVA_HOME and bin path point to a valid JDK home directory. Otherwise, ignore this step.
+2. Open VS Code Command Palette (`Ctrl+Shift+P`), and run command `"Java: Update project configuration"` to force the Language Support extension to resolve the project relationship again.
+3. If the step above doesn't work, please try to clean the stale workspace cache. Close all opened VS Code windows first and delete the following cache directory:
+- Windows : %APPDATA%\Code\User\workspaceStorage\
+- MacOS : $HOME/Library/Application Support/Code/User/workspaceStorage/
+- Linux : $HOME/.config/Code/User/workspaceStorage/
+4. Try more [troubleshooting guide](https://github.com/redhat-developer/vscode-java/wiki/Troubleshooting) from the Language Support extension.
+
+## Failed to resolve classpath: xxx.
+### Reason:
+In launch mode, Java Debugger will auto resolve the classpath based on the given `"mainClass"` and `"projectName"`, if the mainClass or projectName is not correctly configured, it probably throws this kind of error.
+
+### Try:
+1. Check the mainClass is correct. The illegal class name is like `"C:\todoapp\main\src\org\microsoft\app\Main.java"`. The debugger doesn't accept the file name as mainClass. It should be `"${packageName}.${className}"` with `"."` seperated, e.g. `"org.microsoft.app.Main"`.
+2. Check the projectName is correct. The projectName is not always same as the workspace folder name. Usually the project name is read from the `"projectDescription/name"` field in the `".project"` file, or the `"artificatId"` field in the `"pom.xml"`.
+3. If the problem still exists, please use the auto generation feature provided by Java Debugger. Remove the existing launch.json file, and click F5, then Java Debugger will auto resolve the debug configurations into a new generated launch.json.

```

Comment:
If the problem persists, please try to use the debugger to regenerate the debug configurations in *launch.json*.

Reply:
Remove the existing *launch.json* file and press F5. The debugger will automatically generate a new *launch.json* with the right debug configurations.